odoo/tools/sql.py

```diff
@@ -403,6 +403,10 @@ def get_depending_views(cr, table, column):
     return cr.fetchall()
 
 
+class ConstraintError(Exception):
```

Comment:
I'm not sure you need to add another exception.

Reply:
The linter complains about capturing `Exception`, and more importantly: I agree! We only need to capture the error happening when the ORM cannot add a constraint with the message to be logged stored on it. Any other exception shouldn't be conflated into this. It already happened to me while developing this patch that a dummy mistake was wrongly reported as if the constraint cannot be added. While in fact it was just a programming error. I'm OK with not adding the extra exception, but to me this is cheap enough and makes the code clearer and less prone to these types of mistakes. Your call :)